### PR TITLE
Define runtime parameters in Python

### PR DESCRIPTION
**Pull request by icui**
# Description

Implements runtime configuration and add pip dependencies.

# Issue Number

#231

# Checklist

Please make sure to check developer documentation on specfem docs. 

[] I ran the code through pre-commit to check style
[] My code passes all the integration tests
[] I have added sufficient unittests to test my changes
[] I have added/updated documentation for the changes I am proposing
[] I have updated CMakeLists to ensure my code builds
[] My code builds across all platforms

### CODE REVIEW

**Review by lsawade — APPROVED** [review]

File: src/specfem2d.cpp [hunk 1]
```diff
@@ -95,7 +95,13 @@ void execute(const std::string &parameter_file, const std::string &default_file,
   //                    Read parameter file
   // --------------------------------------------------------------
   auto start_time = std::chrono::system_clock::now();
-  specfem::runtime_configuration::setup setup(parameter_file, default_file);
+  specfem::runtime_configuration::setup setup(parameter_file, default_file,
+#ifdef SPECFEMPP_BINDING_PYTHON
```
Comment (Rohit-Kakodkar) [inline src/specfem2d.cpp:99]: This becomes tough to read IMO.

Comment (icui) [reply src/specfem2d.cpp:99]: This should be changed after we template YAML::Node

**Review by Rohit-Kakodkar — CHANGES_REQUESTED** [review]
See comments

**Review by Rohit-Kakodkar — APPROVED** [review]
LGTM

**Pull request merged**